bliss/base_detect/base_detect.py

```diff
@@ -0,0 +1,415 @@
+        if c:
+            LOGGER.info(f"{sum(dt):10.2f} {'-':>10s} {'-':>10s}  Total")
+
+    def fuse(self):  # fuse model Conv2d() + BatchNorm2d() layers
+        LOGGER.info("Fusing layers... ")
+        for m in self.model.modules():
+            if isinstance(m, (Conv)) and hasattr(m, "bn"):  # noqa: WPS421
+                m.conv = fuse_conv_and_bn(m.conv, m.bn)  # update conv
+                delattr(m, "bn")  # remove batchnorm    # noqa: WPS421
+                m.forward = m.forward_fuse  # update forward
+        self.info()
+        return self
+
+    def info(self, verbose=False, img_size=640):  # print model information
+        model_info(self, verbose, img_size)
+
+    def _apply(self, fn):
+        # Apply to(), cpu(), cuda(), half() to model tensors that are not
+        # parameters or registered buffers
+        self = super()._apply(fn)  # pylint: disable=W0642  # noqa: WPS117
+        m = self.model[-1]  # Detect()
+        if isinstance(m, (Detect)):
+            m.stride = fn(m.stride)
+            m.grid = list(map(fn, m.grid))
+            if isinstance(m.anchor_grid, list):
+                m.anchor_grid = list(map(fn, m.anchor_grid))
+        return self
+
+
+class DetectionModel(BaseModel):
+    # pylint: disable=W1203
+    # YOLOv5 detection model
+    def __init__(
+        self, cfg="yolov5s.yaml", ch=3, nc=None, anchors=None
+    ):  # model, input channels, number of classes
+        super().__init__()
+        if isinstance(cfg, dict):
+            self.yaml = cfg  # model dict
+        else:  # is *.yaml
+            import yaml  # for torch hub  # pylint: disable=C0415
+
+            self.yaml_file = Path(cfg).name
+            with open(cfg, encoding="ascii", errors="ignore") as f:
+                self.yaml = yaml.safe_load(f)  # model dict
+
+        # Define model
+        # input channels
+        ch = self.yaml["ch"] = self.yaml.get("ch", ch)  # noqa: WPS429
+        if nc and nc != self.yaml["nc"]:
+            LOGGER.info(f"Overriding model.yaml nc={self.yaml['nc']} with nc={nc}")
+            self.yaml["nc"] = nc  # override yaml value
+        if anchors:
+            LOGGER.info(f"Overriding model.yaml anchors with anchors={anchors}")
+            self.yaml["anchors"] = round(anchors)  # override yaml value
+        self.model, self.save = parse_model(deepcopy(self.yaml), ch=[ch])  # model, savelist
+        self.names = [str(i) for i in range(self.yaml["nc"])]  # default names
+        self.inplace = self.yaml.get("inplace", True)
+
+        # Build strides, anchors
+        m = self.model[-1]  # Detect()
+        if isinstance(m, (Detect)):
+            s = 256  # 2x min stride
+            m.inplace = self.inplace
+            forward = lambda x: self.forward(x)  # pylint: disable=C3001, W0108  # noqa: E731
+
+            # NOTE: 5-D input
+            d = 2  # set d = 2 in 5-band case
+            m.stride = torch.tensor([s / x.shape[-2] for x in forward(torch.zeros(1, ch, d, s, s))])
```

Comment:
This model stride `(m.stride)` had to be overwritten within the constructor of `DetectionModel` - it was originally passing a 4-D zero-tensor of size `(1 x ch x s x s)` to set the stride. Other than this, all other changes that would enable us to use a 5-D input are within `parse_model` and within our `base_config.yaml`. 

If we import `DetectionModel` from yolo, it would be passing in this 4-D zero-tensor input to our 3-D convolution head, which would throw an error during construction. Is there some clever way to get around this without having to add the constructor to our codebase?